src/main/java/org/broadinstitute/hellbender/utils/read/GoogleGenomicsReadToGATKReadAdapter.java

```diff
@@ -136,6 +141,76 @@ public void setPosition( final String contig, final int start ) {
     }
 
     @Override
+    public void unmap() {
+        if (genomicsRead.getAlignment() != null) {
+            final boolean isReverse = genomicsRead.getAlignment().getPosition().getReverseStrand();
+            genomicsRead.setAlignment(null);
+            if (isReverse) { // need to reverse-complement the bases and reverse the qualities.
+                reverseComplement();
+            }
+            genomicsRead.setProperPlacement(false);
+        }
+    }
+
+    /**
+     * Revert and revert complement sequences whose order depend on whether the alignment is on the forward
+     * or negative backward strand.
+     */
+    private void reverseComplement() {
+        if (genomicsRead.getAlignedSequence() != null) {
+            genomicsRead.setAlignedSequence(SequenceUtil.reverseComplement(genomicsRead.getAlignedSequence()));
+        }
+        if (genomicsRead.getAlignedQuality() != null) {
+            genomicsRead.setAlignedQuality(CollectionUtils.reverse(genomicsRead.getAlignedQuality(), ArrayList<Integer>::new));
+        }
+        for (final String tag : SAMRecord.TAGS_TO_REVERSE_COMPLEMENT) {
+            final Object o = genomicsRead.get(tag);
+            if (o != null) {
+                if (o instanceof byte[]) {
+                    final byte[] newBytes = ((byte[]) o).clone();
+                    SequenceUtil.reverseComplement(newBytes);
+                    genomicsRead.set(tag, newBytes);
+                } else if (o instanceof CharSequence) {
+                    genomicsRead.set(tag, SequenceUtil.reverseComplement(o.toString()));
+                } else {
+                    throw new UnsupportedOperationException("cannot handle tag reverse-complement on type: " + o.getClass());
+                }
+            }
+        }
+        for (final String tag : SAMRecord.TAGS_TO_REVERSE) {
+            final Object o = genomicsRead.get(tag);
+            if (o != null) {
+                if (o.getClass().isArray()) {
+                    final int length = Array.getLength(o);
+                    final Object newO = Array.newInstance(o.getClass().getComponentType(), length);
+                    for (int i = 0; i < length; i++) {
+                        Array.set(newO, i, Array.get(o, length - i - 1));
+                    }
+                    genomicsRead.set(tag, newO);
+                } else if (o instanceof CharSequence) {
+                    genomicsRead.set(tag, StringUtils.reverse(o.toString()));
+                } else {
+                    throw new UnsupportedOperationException("cannot handle tag reverse on type: " + o.getClass());
+                }
+            }
+        }
```

Comment:
Ouch -- you will definitely need unit tests in `GATKReadAdaptersUnitTest` for this...